sdk/ml/azure-ai-ml/azure/ai/ml/entities/_assets/_artifacts/featureset.py

```diff
@@ -0,0 +1,145 @@
+        :type materialization_settings: ~azure.ai.ml.entities.MaterializationSettings
+        :param kwargs: A dictionary of additional configuration parameters.
+        :type kwargs: dict
+        """
+        super().__init__(
+            name=name,
+            version=version,
+            description=description,
+            tags=tags,
+            properties=properties,
+            **kwargs,
+        )
+        self.entities = entities
+        self.specification = specification
+        self.stage = stage
+        self.materialization_settings = materialization_settings
+        self.latest_version = None
+
+    def _to_rest_object(self) -> FeaturesetVersion:
+        featureset_version_properties = FeaturesetVersionProperties(
+            description=self.description,
+            properties=self.properties,
+            tags=self.tags,
+            entities=self.entities,
+            materialization_settings=self.materialization_settings._to_rest_object()
+            if self.materialization_settings
+            else None,
+            specification=self.specification._to_rest_object(),
+            stage=self.stage,
+        )
+        return FeaturesetVersion(properties=featureset_version_properties)
+
+    @classmethod
+    def _from_rest_object(cls, featureset_rest_object: FeaturesetVersion) -> "Featureset":
+        if not featureset_rest_object:
+            return None
+        featureset_rest_object_details: FeaturesetVersionProperties = featureset_rest_object.properties
+        arm_id_object = get_arm_id_object_from_id(featureset_rest_object.id)
+        featureset = Featureset(
+            id=featureset_rest_object.id,
+            name=arm_id_object.asset_name,
+            version=arm_id_object.asset_version,
+            description=featureset_rest_object_details.description,
+            tags=featureset_rest_object_details.tags,
+            properties=featureset_rest_object_details.properties,
+            entities=featureset_rest_object_details.entities,
+            materialization_settings=MaterializationSettings._from_rest_object(
+                featureset_rest_object_details.materialization_settings
+            ),
+            specification=FeaturesetSpecification._from_rest_object(featureset_rest_object_details.specification),
+            stage=featureset_rest_object_details.stage,
+        )
+        return featureset
+
+    @classmethod
+    def _from_container_rest_object(cls, rest_obj: FeaturesetContainer) -> "Featureset":
+        rest_object_details: FeaturesetContainerProperties = rest_obj.properties
+        arm_id_object = get_arm_id_object_from_id(rest_obj.id)
+        featureset = Featureset(
+            name=arm_id_object.asset_name,
+            description=rest_object_details.description,
+            tags=rest_object_details.tags,
+            properties=rest_object_details.properties,
+            entities=[],
+            specification=FeaturesetSpecification(),
```

Comment:
Should specification be null for the container?

Reply:
Its the container call, when we don't have version. it will show empty for entities and specification. 
latest version is the property which will show the latest version available in the container. If user want to see the latest version, they get it from this call

Reply:
Got it. Thanks.